### DISCUSSION TITLE
Ability to run MediaCreationTool without a logged in user session?

### DISCUSSION THREAD

**Discussion by smood922** [AveYo/MediaCreationTool.bat#360] [Q&A]
I'd like to use my RMM to run MCT completely non-interactively: no user logged in, and running as SYSTEM. When I attempt that currently, the MCT exe runs but hangs before beginning the download. Has anyone successfully pulled this off?

**Comment by nondetect** [comment]
Are you tried the way what discribed in "[Simple deployment](https://github.com/AveYo/MediaCreationTool.bat#simple-deployment)" ?

**Comment by smood922** [comment]
Yes, that's what I'm following.

When I execute it in the logged in user context, it runs and completes, bit when I run it as system it hangs.

In user context there's one UAC prompt to accept. I suspect that's where it's hanging on system since it's not interactive.

**Comment by ccalgher** [comment]
So, I've been hard stuck on this as well, and in my troubleshooting I decided to try and view what running it with the gui would look like under the system account. Obviously you can't login to the machine as the system account, but you can run gui-based files in the context of the system account in a regular user desktop session by using psexec, with the -i -s -h switches. What I found when letting it run under the system account was this: 
![2024-09-05_10-58-08](https://github.com/user-attachments/assets/591cd491-571e-4466-950d-b9e8b36a5aa6)
it seems like MCT by default tries to open the file to save the iso on the desktop of whatever user is running the tool, but sadly that does not exist for the system account, and it's throwing an extra prompt/thing to click through that AveYo's batch file isn't setup to click through I guess?

Strangely though, if you click through the menus in the 'save as' window there and manually start to save it somewhere else, but then cancel and re-do the whole process of running it under the system account, it goes through the gui just fine because it's no longer immediately getting a warning pop up for a non-existent folder, it will default to where you tried saving it during the last go-around.

**Reply by smood922** [reply]
Nice work! That last part seems like a possible path forward. Have you tried procmon'ing the executable to see if it's writing that path anywhere that we could pre-populate in the batch script?

**Reply by ccalgher** [reply]
I did find the line in the batch script that pre-populates the save location, and it does enter it in as expected, however that warning still pops up. That warning pops up IMMEDIATELY when the "save as" explorer window is opened, and the batch script I don't believe knows how to click through it. I believe the MCT tool itself defaults to the documents folder, or windows itself does. I'm not sure if there's a registry key somewhere that could change that though, but that's a possibility.

Edit: I've also opened this up as an issue too

**Reply by 3p1st3m1c** [reply]
I've also tried running a scheduled task with the highest privilege "-Execute "C:\Windows\System32\cmd.exe" -Argument "/c start /wait C:\Temp\Win11\auto 23H2 MediaCreationTool.bat" whether logged in as the local Admin or running as a system account.  It gets stuck and does nothing.  I am almost thinking the only way to do this is to use PowerAutomate, login as a local Admin and then right mouse click on the batch file, and select RUN AS ADMINISTRATOR.  That's the only way I can get this to work.  Yes.  I want to automate this with the RMM I use as well.

**Comment by 3p1st3m1c** [comment]
I've given up on trying to automate this via RMM.  Perhaps the batch file needs to be rewritten completely in Powershell...
